chia/full_node/full_node.py

```diff
@@ -1003,6 +1003,9 @@ async def respond_block(
         if peak.height % 1000 == 0 and not self.sync_store.get_sync_mode():
             await self.sync_store.clear_sync_info()  # Occasionally clear sync peer info
         self._state_changed("block")
+        record = self.blockchain.block_record(block.header_hash)
```

Comment:
This will slow down the node, right? Could we do it every sub-epoch instead of every block?

Reply:
it happenes only when sub epoch ends though

Reply:
the block_record call is a call to the cache

Reply:
I meant the next line, which creates the segments

Reply:
that happens only at the end of a sub epoch

Reply:
@mariano54 i can move it to the end of short_sync_backtrack that might be a better place to call it

Reply:
Oh yeah I can't believe I missed that, you're right.

Reply:
No need to move it into backtrack. That still get's called once per block, so it's the same.

Reply:
maybe we should do some better concurrency here but im not sure how to do that without slowing down things under 
blockchain.lock

Reply:
i can probably clean all the requests if we go in to long sync, i think that should work without slowing down gossip

Reply:
it is probably isn't that bad, it will just overwrite but it will use the correct segment just redundant work

Reply:
@mariano54 i added tracking for the task, and cancelling if they stay for the next sub_epoch/long sync